Nightly feature fails to build on 2018-05-28 nightly

```console
$ cargo build --features nightly

   Compiling unicode-xid v0.1.0
   Compiling proc-macro2 v0.4.3
error[E0599]: no function or associated item named `empty` found for type `proc_macro::TokenStream` in the current scope
  --> src/unstable.rs:45:34
   |
45 |             TokenStream::Nightly(proc_macro::TokenStream::empty())
   |                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function or associated item not found in `proc_macro::TokenStream`
```

https://github.com/rust-lang/rust/pull/51073

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -103,8 +103,13 @@ impl TokenStream {
     }
 
     /// Returns an empty `TokenStream` containing no token trees.
+    pub fn new() -> TokenStream {
+        TokenStream::_new(imp::TokenStream::new())
+    }
+
+    #[deprecated(since = "0.4.4", note = "please use TokenStream::new")]
     pub fn empty() -> TokenStream {
-        TokenStream::_new(imp::TokenStream::empty())
+        TokenStream::new()
     }
 
     /// Checks if this `TokenStream` is empty.

--- a/src/stable.rs
+++ b/src/stable.rs
@@ -25,7 +25,7 @@ pub struct TokenStream {
 pub struct LexError;
 
 impl TokenStream {
-    pub fn empty() -> TokenStream {
+    pub fn new() -> TokenStream {
         TokenStream { inner: Vec::new() }
     }
 

--- a/src/unstable.rs
+++ b/src/unstable.rs
@@ -40,11 +40,11 @@ fn mismatch() -> ! {
 }
 
 impl TokenStream {
-    pub fn empty() -> TokenStream {
+    pub fn new() -> TokenStream {
         if nightly_works() {
-            TokenStream::Nightly(proc_macro::TokenStream::empty())
+            TokenStream::Nightly(proc_macro::TokenStream::new())
         } else {
-            TokenStream::Stable(stable::TokenStream::empty())
+            TokenStream::Stable(stable::TokenStream::new())
         }
     }
 

